sktime/performance_metrics/detection/_randindex.py

```diff
@@ -0,0 +1,92 @@
+from collections import Counter
+
+from sktime.performance_metrics.detection._base import BaseDetectionMetric
+
+
+class RandIndex(BaseDetectionMetric):
+    """Rand Index metric for comparing event detection results."""
+
+    _tags = {
+        "object_type": ["metric_detection", "metric"],
+        "scitype:y": "segments",
+        "requires_X": False,
+        "requires_y_true": True,
+        "lower_is_better": False,  # Higher Rand Index is better
+    }
+
+    def _evaluate(self, y_true, y_pred, X=None):
+        """Calculate Rand Index between true and predicted segments.
+
+        Parameters
+        ----------
+        y_true : pd.DataFrame
+            Ground truth segments
+        y_pred : pd.DataFrame
+            Predicted segments
+        X : pd.DataFrame, optional (default=None)
+            Not used
+
+        Returns
+        -------
+        float
+            Rand Index score between 0.0 and 1.0
+        """
+        # Convert DataFrame format to segments list
+        ground_truth_segments = [
+            (row.Index, row.Index + 1) for row in y_true.itertuples()
+        ]
+        predicted_segments = [(row.Index, row.Index + 1) for row in y_pred.itertuples()]
+
+        # 1) Determine total length (N)
+        total_length = max(
+            max(end for _, end in ground_truth_segments),
+            max(end for _, end in predicted_segments),
+        )
+
+        # 2) Create label arrays: ground_truth_labels[i] = cluster_id of i in ground-truth  # noqa: E501
+        #                         predicted_labels[i]    = cluster_id of i in predicted
+        ground_truth_labels = [None] * total_length
+        predicted_labels = [None] * total_length
+
+        # Assign cluster IDs for ground truth
+        # We can just use an incremental ID for each segment
+        for cluster_id, (start, end) in enumerate(ground_truth_segments):
```

Comment:
The segmentation can be labelled, in wihch case the data frame would contain a column `labels`. This assumes it is unlabelled?